### DISCUSSION TITLE
Automatic Battery Management for Mobile Use -- Auto-Start & Auto-Graceful Stop?

### DISCUSSION THREAD

**Discussion by turnkit** [JosephHewitt/wardriver_rev3#101] [Ideas]
Would you consider adding some battery management and a way to add a 18650 to an optional add-on board?

What I would love to see is a way to plug the device into a USB-C charge port (from my car cab power adapter) such that I could drive around and then when I turn the ignition power off the device stays on for a while longer (15 minutes, or until power gets down to 10%?) and then sends a graceful shutdown to the system.

Optionally the device could stay on as long as it could until the power reached a pre-set percentage of life yet.

Also, if not obvious, the device would take the start up of the ignition (i.e. power plugged in) as a start up signal so the Wardriver would start up when the car started.

The whole goal is to automate the logging process such that I can plug-and-forget, and do so in a way that will allow good capture lengths, even for some time after I've parked.  Or am I missing that this is sort of already happening?

**Comment by JosephHewitt** [comment]
Nice idea. This isn't something I want to implement on the current iteration (the rev3) but I am currently working on the hardware of the new revision, so this is definitely something I will keep in mind.

I've used this [Adafruit Board](https://www.adafruit.com/product/4755) in other projects and quite like the way it works; it is designed for solar but also has USB-C and a barrel connector, plus a connector for a battery. I used an 18650 with it, but had to add my own battery holder.

This board has some pins which go high/low to indicate if it is being powered externally, and one to indicate if the battery is charged.

It would be quite easy to add some pins on the edge of a new PCB which something like this can be connected to. The code could detect a switch from powered to battery, then start a count down.

I don't think it would be possible to fully power down the system without making this board an integral non-optional component, but by putting everything into deep sleep it should consume such a low amount of power that it would last for days/weeks.
